tests/unit/modules/default/utils_spec.js

```diff
@@ -0,0 +1,111 @@
+
+			describe("When using cors proxy", () => {
+				Object.defineProperty(global, "location", {
+					value: {
+						host: locationHost,
+						protocol: locationProtocol
+					}
+				});
+
+				test("Calls correct URL once", async () => {
+					const urlToCall = "http://www.test.com/path?param1=value1";
+					url = urlToCall;
+
+					await performWebRequest(url, "json", true);
+
+					expect(fetchMock.mock.calls.length).toBe(1);
+					expect(fetchMock.mock.calls[0][0]).toBe(`${locationProtocol}//${locationHost}/cors?url=${urlToCall}`);
+				});
+
+				test("Sends correct headers", async () => {
+					const urlToCall = "http://www.test.com/path?param1=value1";
+					url = urlToCall;
+					const headers = [
+						{ name: "header1", value: "value1" },
+						{ name: "header2", value: "value2" }
+					];
+
+					await performWebRequest(url, "json", true, headers);
+
+					expect(fetchMock.mock.calls.length).toBe(1);
+					expect(fetchMock.mock.calls[0][0]).toBe(`${locationProtocol}//${locationHost}/cors?sendheaders=header1:value1,header2:value2&url=${urlToCall}`);
+				});
+			});
+
+			describe("When not using cors proxy", () => {
+				test("Calls correct URL once", async () => {
+					const urlToCall = "http://www.test.com/path?param1=value1";
+					url = urlToCall;
+
+					await performWebRequest(url);
+
+					expect(fetchMock.mock.calls.length).toBe(1);
+					expect(fetchMock.mock.calls[0][0]).toBe(urlToCall);
+				});
+
+				test("Sends correct headers", async () => {
+					const urlToCall = "http://www.test.com/path?param1=value1";
+					url = urlToCall;
+					const headers = [
+						{ name: "header1", value: "value1" },
+						{ name: "header2", value: "value2" }
+					];
+
+					await performWebRequest(url, "json", false, headers);
+
+					const expectedHeaders = { headers: { header1: "value1", header2: "value2" } };
+					expect(fetchMock.mock.calls.length).toBe(1);
+					expect(fetchMock.mock.calls[0][1]).toStrictEqual(expectedHeaders);
+				});
+			});
+
+			describe("When receiving json format", () => {
+				test("Returns undefined when no data is received", async () => {
+					const response = await performWebRequest(url);
+
+					expect(response).toBe(undefined);
+				});
+
+				test("Returns object when data is received", async () => {
+					fetchResponse = new Response('{"body": "some content"}');
+
+					const response = await performWebRequest(url);
+
+					expect(response.body).toBe("some content");
+				});
+
+				test("Returns expected headers when data is received", async () => {
+					fetchResponse = new Response('{"body": "some content"}', { headers: { header1: "value1", header2: "value2" } });
+
+					const response = await performWebRequest(url, "json", false, undefined, ["header1"]);
+
+					expect(response.headers.length).toBe(1);
+					expect(response.headers[0].name).toBe("header1");
+					expect(response.headers[0].value).toBe("value1");
+				});
+			});
+		} else {
+			test("Always ok, need one test", () => {});
```

Comment:
so why exactly dont we need a test for node < 18 ?

Reply:
Because they don't run 👀  It boils down to the fetch-method (in much the same way as I struggled with fetch in https://github.com/MichMich/MagicMirror/pull/2957#issuecomment-1290958862). When I tested with node `16.17.1` I got the following error:

<img width="694" alt="Skjermbilde 2022-11-07 kl  22 14 32" src="https://user-images.githubusercontent.com/34011212/200416879-0bd2c335-a824-44b5-8dcc-bc28e5c9114f.png">

It seems the fetch-related types are not defined. According to the [docs](https://developer.mozilla.org/en-US/docs/Web/API/fetch#browser_compatibility) fetch was introduced in Node 18, so it makes sense. Fetch was introduced to the weather module in https://github.com/MichMich/MagicMirror/pull/2935. I figured since fetch was already implemented I wouldn't bother with reimplementing another kind of fetch.